Building index params check for Annoy

**Describe the bug**
Expected build failed status returned if `n_trees` set an invalid valid value like: `-1/string/empty/..`
```
_______________________________________ TestCreateIndexParamsInvalid.test_create_index_with_invalid_index_params[get_index119] _______________________________________

self = <test_index.TestCreateIndexParamsInvalid object at 0x7fa76c43a8d0>, connect = <milvus.client.stub.Milvus object at 0x7fa7ef5271d0>
collection = 'test_DaDSgPgi', get_index = {'index_param': {'n_trees': 'String'}, 'index_type': <IndexType: ANNOY>}

    @pytest.mark.level(1)
    def test_create_index_with_invalid_index_params(self, connect, collection, get_index):
        index_param = get_index["index_param"]
        index_type = get_index["index_type"]
        logging.getLogger().info(get_index)
        # status, ids = connect.add_vectors(collection, vectors)
        if (not index_type) or (not isinstance(index_type, IndexType)):
            with pytest.raises(Exception) as e:
                status = connect.create_index(collection, index_type, index_param)
        else:
            status = connect.create_index(collection, index_type, index_param)
>           assert not status.OK()
E           AssertionError: assert not True
E            +  where True = <bound method Status.OK of Status(code=0, message='Build index successfully!')>()
E            +    where <bound method Status.OK of Status(code=0, message='Build index successfully!')> = Status(code=0, message='Build index successfully!').OK

test_index.py:1794: AssertionError
```
**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
- Hardware/Softward conditions (OS, CPU, GPU, Memory)
- Method of installation (Docker, or from source)
- Milvus version (v0.3.1, or v0.4.0)
- Milvus configuration (Settings you made in `server_config.yaml`)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -119,6 +119,7 @@ Please mark all change in change log and use the issue from GitHub
 -   \#1598 Server down during mixed operations
 -   \#1601 External link bug in HTTP doc
 -   \#1609 Refine Compact function
+-   \#1808 Building index params check for Annoy
 
 ## Feature
 -   \#216 Add CLI to get server info

--- a/core/src/index/knowhere/knowhere/index/vector_index/ConfAdapter.cpp
+++ b/core/src/index/knowhere/knowhere/index/vector_index/ConfAdapter.cpp
@@ -299,9 +299,9 @@ BinIVFConfAdapter::CheckTrain(Config& oricfg, const IndexMode mode) {
 
 bool
 ANNOYConfAdapter::CheckTrain(Config& oricfg, const IndexMode mode) {
-    static int64_t MIN_NTREES = 0;
+    static int64_t MIN_NTREES = 1;
     // too large of n_trees takes much time, if there is real requirement, change this threshold.
-    static int64_t MAX_NTREES = 16384;
+    static int64_t MAX_NTREES = 1024;
 
     CheckIntByRange(knowhere::IndexParams::n_trees, MIN_NTREES, MAX_NTREES);
 

--- a/core/src/index/thirdparty/annoy/src/annoylib.h
+++ b/core/src/index/thirdparty/annoy/src/annoylib.h
@@ -967,7 +967,7 @@ public:
       memcpy(_get(_n_nodes + (S)i), _get(_roots[i]), _s);
     _n_nodes += _roots.size();
 
-    if (_verbose) showUpdate("has %d nodes\n", _n_nodes);
+    if (_verbose) showUpdate("has %ld nodes\n", _n_nodes);
     
     if (_on_disk) {
       _nodes = remap_memory(_nodes, _fd, _s * _nodes_size, _s * _n_nodes);
@@ -1105,7 +1105,7 @@ public:
     _loaded = true;
     _built = true;
     _n_items = m;
-    if (_verbose) showUpdate("found %lu roots with degree %d\n", _roots.size(), m);
+    if (_verbose) showUpdate("found %lu roots with degree %ld\n", _roots.size(), m);
     return true;
   }
 
@@ -1144,7 +1144,7 @@ public:
     _loaded = true;
     _built = true;
     _n_items = m;
-    if (_verbose) showUpdate("found %lu roots with degree %d\n", _roots.size(), m);
+    if (_verbose) showUpdate("found %lu roots with degree %ld\n", _roots.size(), m);
     return true;
   }
 
@@ -1215,7 +1215,7 @@ protected:
       }
       
       _nodes_size = new_nodes_size;
-      if (_verbose) showUpdate("Reallocating to %d nodes: old_address=%p, new_address=%p\n", new_nodes_size, old, _nodes);
+      if (_verbose) showUpdate("Reallocating to %ld nodes: old_address=%p, new_address=%p\n", new_nodes_size, old, _nodes);
     }
   }
 
@@ -1266,7 +1266,7 @@ protected:
         bool side = D::side(m, n->v, _f, _random);
         children_indices[side].push_back(j);
       } else {
-        showUpdate("No node for index %d?\n", j);
+        showUpdate("No node for index %ld?\n", j);
       }
     }
 
@@ -1317,7 +1317,7 @@ protected:
     std::priority_queue<pair<T, S> > q;
 
     if (search_k <= 0) {
-      search_k = n * _roots.size();
+      search_k = std::max(n * _roots.size(), (size_t )_n_items * 5 / 100);
     }
 
     for (size_t i = 0; i < _roots.size(); i++) {

--- a/core/src/utils/ValidationUtil.cpp
+++ b/core/src/utils/ValidationUtil.cpp
@@ -265,6 +265,13 @@ ValidationUtil::ValidateIndexParams(const milvus::json& index_params,
             }
             break;
         }
+        case (int32_t)engine::EngineType::ANNOY: {
+            auto status = CheckParameterRange(index_params, knowhere::IndexParams::n_trees, 1, 1024);
+            if (!status.ok()) {
+                return status;
+            }
+            break;
+        }
     }
     return Status::OK();
 }
@@ -297,6 +304,13 @@ ValidationUtil::ValidateSearchParams(const milvus::json& search_params,
         }
         case (int32_t)engine::EngineType::HNSW: {
             auto status = CheckParameterRange(search_params, knowhere::IndexParams::ef, topk, 4096);
+            if (!status.ok()) {
+                return status;
+            }
+            break;
+        }
+        case (int32_t)engine::EngineType::ANNOY: {
+            auto status = CheckParameterRange(search_params, knowhere::IndexParams::search_k, std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::max());
             if (!status.ok()) {
                 return status;
             }

--- a/core/src/utils/ValidationUtil.cpp
+++ b/core/src/utils/ValidationUtil.cpp
@@ -27,6 +27,7 @@
 #include <fiu-local.h>
 #include <algorithm>
 #include <cmath>
+#include <limits>
 #include <regex>
 #include <string>
 
@@ -310,7 +311,8 @@ ValidationUtil::ValidateSearchParams(const milvus::json& search_params,
             break;
         }
         case (int32_t)engine::EngineType::ANNOY: {
-            auto status = CheckParameterRange(search_params, knowhere::IndexParams::search_k, std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::max());
+            auto status = CheckParameterRange(search_params, knowhere::IndexParams::search_k,
+                                              std::numeric_limits<int64_t>::min(), std::numeric_limits<int64_t>::max());
             if (!status.ok()) {
                 return status;
             }

